xwiki-platform-core/xwiki-platform-annotation/xwiki-platform-annotation-io/src/main/java/org/xwiki/annotation/io/internal/DefaultIOService.java

```diff
@@ -200,12 +207,13 @@ public Collection<Annotation> getAnnotations(String target) throws IOServiceExce
                 return Collections.emptySet();
             }
             for (BaseObject object : objects) {
-                // if it's not on the required target, ignore it
-                if (object == null || !localTargetId.equals(object.getStringValue(Annotation.TARGET_FIELD))) {
-                    continue;
-                }
                 // use the object number as annotation id
-                result.add(loadAnnotationFromObject(object, deprecatedContext));
+                if (object != null) {
+                    String targetField = object.getStringValue(Annotation.TARGET_FIELD);
+                    if (Objects.equals(localTargetId, targetField) || Objects.equals("", targetField)) {
```

Comment:
Shouldn't annotations with empty target only be loaded if the target is actually the document (i.e., its content) and not an XObject property reference? With the current code, annotations that are for the content will also be loaded for every object property.

Wouldn't it be safer to use [`StringUtils.isBlank(targetField)`](https://commons.apache.org/proper/commons-lang/javadocs/api-2.6/org/apache/commons/lang/StringUtils.html#isBlank%28java.lang.String%29) to also catch, e.g., `null` values or user-edited objects where the user unintentionally inserted a space?

Reply:
> Shouldn't annotations with empty target only be loaded if the target is actually the document (i.e., its content) and not an XObject property reference? With the current code, annotations that are for the content will also be loaded for every object property.

I'm probably missing something here. What's the usecase you have in mind?

Reply:
According to the code and [the forum discussion](https://forum.xwiki.org/t/annotations-and-the-target-property/12187/3), the target is either a) a document reference, in which case the annotation is for the document's content or b) an object property reference, in which case the annotations is for an object property. My understanding is that for the first case, you're now using the empty string as reference. In the second case, the original target reference is still stored and annotations still break after moving, but my understanding is that we have currently no UI for that, anyways, so at least with the standard UI that won't be visible. However, when whatever code would load annotations for an object property, your changed code would now not only return annotations for that object property but also for the document's content, which, even if unused currently, is a regression.